cmd/bnc-config/skl.go

```diff
@@ -0,0 +1,190 @@
+)
+
+var geo = ellipsoid.Init("WGS84", ellipsoid.Degrees, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
+
+// genetrate skeleton file for a site,
+// based on the receiver, firmware, and antenna metadata at a given time (e.g. now)
+func skeleton(code string, set *meta.Set, ts int64) (content string, err error) {
+	var mark meta.Mark
+
+	defer func() {
+		if err != nil {
+			// if any error occurs, compose generic header + distribution comment as the result
+			// however, still keep the error
+			content = generic_header
+			if mark.Network == "LI" {
+				content += linzComment
+			} else {
+				content += geonetComment
+			}
+		}
+	}()
+
+	mark, ok := set.Mark(code)
+	if !ok {
+		err = fmt.Errorf("no mark found for %s", code)
+		return
+	}
+	if !inWindow(ts, mark.Span) {
+		err = fmt.Errorf("no valid mark found for this time period for %s", code)
+		return
+	}
+
+	receivers := set.DeployedReceivers()
+	var dr meta.DeployedReceiver
+	for _, r := range receivers {
+		if r.Mark == code && inWindow(ts, r.Span) {
+			dr = r
+			break
+		}
+	}
+
+	if dr.Span.Start.IsZero() { // empty value
+		err = fmt.Errorf("no effective deployed receiver version found for this time for %s", mark.Code)
+		return
+	}
+
+	x, y, z := geo.ToECEF(mark.Latitude, mark.Longitude, mark.Elevation)
+	firmwares := set.FirmwareHistory()
+	var ifirm meta.FirmwareHistory
+	for _, f := range firmwares {
+		if dr.Model == f.Model && dr.Serial == f.Serial && inWindow(ts, f.Span) {
+			ifirm = f
+			break
+		}
+	}
+
+	if ifirm.Span.Start.IsZero() {
+		err = fmt.Errorf("no effective firmware version found for this time for %s", mark.Code)
+		return
+	}
+
+	antennas := set.InstalledAntennas()
+	var ia meta.InstalledAntenna
+	for _, a := range antennas {
+		if a.Mark == code && inWindow(ts, a.Span) {
+			ia = a
+			break
+		}
+	}
+
+	if ia.Span.Start.IsZero() {
+		err = fmt.Errorf("no effective installed antenna found for this time for %s", mark.Code)
+		return
+	}
+
+	radomes := set.InstalledRadomes()
+	var rad meta.InstalledRadome
+	for _, r := range radomes {
+		if r.Mark == code && inWindow(ts, r.Span) {
+			rad = r
+			break
+		}
+	}
+	// radome is optional, can be nil
+	var radome string
+	if rad.Span.Start.IsZero() {
+		radome = "NONE"
+	} else {
+		radome = rad.Model
+	}
+
+	var monument meta.Monument
+	for _, n := range set.Monuments() {
+		if n.Mark == code && inWindow(ts, n.Span) {
+			monument = n
+			break
+		}
+	}
+
+	var domesNumber string
+	if monument.Span.Start.IsZero() {
+		domesNumber = "UNKNOWN"
+	} else {
+		domesNumber = monument.DomesNumber
+	}
+
+	content = fmt.Sprintf(skeletonFormat,
+		fmt.Sprintf("%s00NZL", code),       // MARKER NAME
+		domesNumber,                        // MARKER NUMBER
+		dr.Serial, dr.Model, ifirm.Version, // REC # / TYPE / VERS
+		ia.Serial, ia.Model, radome, //ANT # / TYPE
+		x, y, z, //APPROX POSITION XYZ
+		ia.Offset.Vertical, ia.Offset.East, ia.Offset.North) // ANTENNA: DELTA H/E/N
```

Comment:
nitpick: I'm surprised the formatter didn't align the comments

Reply:
Yeah tried to align them manually. And it bounced back like this right after I clicked "save".

Reply:
huh weird. The online Go Playground doesn't do this (it aligns comments). Ah well.